pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -601,17 +655,49 @@ func getSpecEnv(envs []corev1.EnvVar, name string) string {
 	return ""
 }
 
-func getSmartRouterProtocol(cr *api.KieApp) string {
+func getSmartRouterTermination(cr *api.KieApp) (termination string, protocol string) {
 	if cr.Status.Applied.Objects.SmartRouter != nil {
-		if len(cr.Status.Applied.Objects.SmartRouter.Protocol) == 0 && cr.Status.Applied.CommonConfig.DisableSsl {
-			return constants.HttpProtocol
-		} else if len(cr.Status.Applied.Objects.SmartRouter.Protocol) == 0 && !cr.Status.Applied.CommonConfig.DisableSsl {
-			return constants.HttpsProtocol
+		if cr.Status.Applied.Objects.SmartRouter.TerminationRoute.EnableEdge {
+			// Edge is always http
+			return constants.EdgeTermination, constants.HttpProtocol
+		} else if cr.Status.Applied.CommonConfig.DisableSsl {
+			return constants.DefaultTermination, constants.HttpProtocol
+		} else {
+			return constants.DefaultTermination, constants.HttpsProtocol
+		}
+	} else {
+		return constants.DefaultTermination, constants.HttpsProtocol
+	}
```

Comment:
There is missing case, when there is no config for smart router but dsiableSsl parameter is used. Then there can be wrongly used configuration.

Reply:
Refactored to fallback to the legacy behaviour if enableEdge isn't true